### ISSUE TITLE
Load routine documentation: goes image kyoto lanl maven mica

### ISSUE THREAD

**Issue by jameswilburlewis** [spedas/pyspedas#693]
- [ ] Update docstrings for all load routines defined in the above directories, including those in` __init__.py`.  For preferred style and elements to be included, see https://spedas.org/wiki/index.php?title=PySPEDAS_docstring_format
- [ ] Update or add README.md files in each directory, ensuring all load routines are documented
- [ ] Ensure docstrings include example imports/calls (often can be found in README.md files)

If you find something that's not working, or if it's unclear how to document, please open a new iissue, then keep going with the next routine.  We'll come back and pick up any stragglers in another pass; the idea here is to get all the low-hanging fruit as quickly as possible.

**Comment by nickssl** [comment]
Should we follow the google docstring style or the numpy docstring style? 

I believe that it is a good idea to decide on one or the other and then try to use the same docstring style for the whole project. 

Examples follow.

**Comment by nickssl** [comment]
Numpy docstring style. Mostly used by data scientists. 

https://numpydoc.readthedocs.io/en/latest/format.html

Example:

```

def abc(a: int, c = [1,2]):
    """_summary_

    Parameters
    ----------
    a : int
        _description_
    c : list, optional
        _description_, by default [1,2]

    Returns
    -------
    _type_
        _description_

    Raises
    ------
    AssertionError
        _description_
    """
    if a > 10:
        raise AssertionError("a is more than 10")

    return c
```

**Comment by nickssl** [comment]
Google docstring style. Quite popular outside science. 

https://google.github.io/styleguide/pyguide.html

```
def abc(a: int, c = [1,2]):
    """_summary_

    Args:
        a (int): _description_
        c (list, optional): _description_. Defaults to [1,2].

    Raises:
        AssertionError: _description_

    Returns:
        _type_: _description_
    """
    if a > 10:
        raise AssertionError("a is more than 10")

    return c
```

**Comment by xandrd** [comment]
Hi @nickssl and @jameswilburlewis,

I did a little bit of digging, and this is what I found:

 - Eric indeed planned to use numpy documentation style, not reST as I thought before. 
 - The [Sphinx configuration](https://github.com/spedas/pyspedas/blob/master/docs/source/conf.py) explicitly enables numpy docstring style via the most popular and built-in plugin `Napoleon`.
 - The configuration excludes the `__init.py__` files and the privite method (e.g., function that starts with underline `_private_method()`) from consideration. I assume we still what to the document those for anyone who will look into the code directly.
 - I asked ChatGPT to analyze the PySPEDAS code using `AskTheCode` plugin, and it confirmed that most of the documentation uses numpy style. 
 - However, the style is inconsistent. For example, `mms.fgm` routine uses numpy style correctly:
```python
def mms_load_fgm(trange=['2015-10-16', '2015-10-17'], probe='1', data_rate='srvy',
    level='l2', instrument='fgm', datatype='', varformat=None, exclude_format='*rdeltahalf*', varnames=[], suffix='',
    keep_flagged=False, get_support_data=True, time_clip=False, no_update=False,
    available=False, notplot=False, latest_version=False, major_version=False, 
    min_version=None, cdf_version=None, spdf=False, always_prompt=False, no_split_vars=False,
    get_fgm_ephemeris=False):
    """
    Load MMS fluxgate magnetometer data
    
    Parameters
    ----------
        trange : list of str
            time range of interest [start time, end time] with the format
            'YYYY-MM-DD','YYYY-MM-DD'] or to specify more or less than a day 
            ['YYYY-MM-DD/hh:mm:ss','YYYY-MM-DD/hh:mm:ss']
            Default: ['2015-10-16', '2015-10-17']

        probe : str or list of str
            list of probes, valid values for MMS probes are ['1','2','3','4'].
            Default: '1'

    . . .

    Returns
    ----------
        List of tplot variables created.

    Example:
    >>> import pyspedas
    >>> from pytplot import tplot
    >>> fgm_vars = pyspedas.mms.mms_load_fgm(trange=['2015-10-16', '2015-10-17'])
    >>> tplot('mms1_fgm_b_gsm_srvy_l2')

    """
```
while `themis.fgm` routine does not use correct headings (the underlined hyphens are missing). It uses Google style of heading and numpy style of the variable description:
```python
def fgm(trange=['2007-03-23', '2007-03-24'],
        probe='c',
        level='l2',
        suffix='',
        get_support_data=False,
        varformat=None,
        coord=None,
        varnames=[],
        downloadonly=False,
        notplot=False,
        no_update=False,
        time_clip=False):
    """
    This function loads Fluxgate magnetometer (FGM) data

    Parameters:
        trange: list of str
            time range of interest [starttime, endtime] with the format
            'YYYY-MM-DD','YYYY-MM-DD'] or to specify more or less than a day
            ['YYYY-MM-DD/hh:mm:ss','YYYY-MM-DD/hh:mm:ss']

        probe: str or list of str
            Spacecraft probe letter(s) ('a', 'b', 'c', 'd' and/or 'e')

    . . . 

    Returns:
        List of tplot variables created.
    """
```
- Despite this, the rendered documentation looks the same for [mms.fgm](https://pyspedas.readthedocs.io/en/latest/mms.html#pyspedas.mms.fgm) and [themis.fgm](https://pyspedas.readthedocs.io/en/latest/themis.html#pyspedas.themis.fgm). However, you may notice that the return section for `mms.fgm` have these weird blocks. This is because the return section deviates from the style guideline. Actually, it deviates in both cases. But, in case of `mms.fgm` Sphinx recognizes the numpy style, whereas for  `thims.fgm`, it attempts to accommodate the hybrid style. The correct return section must explicitly include the type:
```python
"""
    Returns
    ----------
    list
        List of tplot variables created.
"""
```


To summarize, I agree with @nickssl that one style should be selected, and it seems that the numpy is the preferred. Sphinx is already configured for it, and ChatGPT indicates that most of the code is documented this way. However, there are inconsistency that ideally should be addressed, but somehow Sphinx can work with the hybrid style that we have now.

**Comment by nickssl** [comment]
Sphinx can now handle both Google docstrings and Numpy docstrings. For example: 

https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html

Quoting the above: "The choice between styles is largely aesthetic, but the two styles should not be mixed. Choose one style for your project and be consistent with it."


Personally, I like both styles. If we decide on Numpy, that's great. Google is fine, too. Numpy style is a little longer, Google style is a little wider. I gave full examples for both in previous comments.

Right now most files in pyspedas do not actually follow either Numpy or Google. Many files are close enough to Numpy but not exactly. Other files are a mixture, some files do not follow any style, and other files do not have proper documentation at all. 

We can decide on one or the other and we can fix the older files in the future.

**Comment by nickssl** [comment]
Completed. 

Docstrings for: goes image kyoto lanl maven mica
